includes/process-donation.php

```diff
@@ -1139,51 +1139,45 @@ function give_register_and_login_new_user( $user_data = [] ) {
 /**
  * Get Donation Form User
  *
+ * @since   1.0
+ * @unreleased Do not run validation check for ajax request expect donation validation ajax request.
+ *
  * @param array $valid_data Valid Data.
  *
  * @access  private
- * @since   1.0
- *
  * @return  array|bool
  */
 function give_get_donation_form_user( $valid_data = [] ) {
-
-	// Initialize user.
-	$user      = false;
-	$is_ajax   = defined( 'DOING_AJAX' ) && DOING_AJAX;
-	$post_data = give_clean( $_POST ); // WPCS: input var ok, sanitization ok, CSRF ok.
-
-	if ( $is_ajax ) {
-
-		// Do not create or login the user during the ajax submission (check for errors only).
-		return true;
-	} elseif ( is_user_logged_in() ) {
-
-		// Set the valid user as the logged in collected data.
-		$user = $valid_data['logged_in_user'];
-	} elseif ( true === $valid_data['need_new_user'] || true === $valid_data['need_user_login'] ) {
-
-		// New user registration.
-		if ( true === $valid_data['need_new_user'] ) {
-
-			// Set user.
-			$user = $valid_data['new_user_data'];
-
-			// Register and login new user.
-			$user['user_id'] = give_register_and_login_new_user( $user );
-
-		} elseif ( true === $valid_data['need_user_login'] && ! $is_ajax ) {
-
-			/**
-			 * The login form is now processed in the give_process_donation_login() function.
-			 * This is still here for backwards compatibility.
-			 * This also allows the old login process to still work if a user removes the checkout login submit button.
-			 *
-			 * This also ensures that the donor is logged in correctly if they click "Donation" instead of submitting the login form, meaning the donor is logged in during the donation process.
-			 */
-			$user = $valid_data['login_user_data'];
-
-			// Login user.
+    // Initialize user.
+    $user      = false;
+    $post_data = give_clean($_POST); // WPCS: input var ok, sanitization ok, CSRF ok.
+    $is_ajax   = ! empty($_POST['give_ajax']) ? $post_data['give_ajax'] : 0; // WPCS: input var ok, sanitization ok, CSRF ok.
```

Comment:
@kjohnson I added only this line.

Reply:
@ravinderk could you elaborate as to how this is different?

Reply:
@kjohnson I explained it in the PR description. let me know if looking for more details. I pointed out this line because other were formatting changes

Reply:
> I update logic to run check on all ajax request except donation validation request as we are doing in give_process_form_login function.

I think this could be made more clear in the code. Looking at the code I don't correlate `$post_data[give_ajax]` with a specific kind of request.

Could we update this to be more declarative? Or being an older part of the system at least add some inline documentation as to what and why of this change? If we come across a bug we are going to see the inline comments but may not necessarily dig up the related PR.

Reply:
@kjohnson If you read this `give_get_donation_form_user` function then you find that there is already a comment inside `if` condition.

Reply:
> // Do not create or login the user during the ajax submission (check for errors only).

Reply:
Ok, so then could we update the variable name? `$is_ajax` is confusing since `DOING_AJAX` could be true while `$post_data['give_ajax']` could be false, in which case `$is_ajax` would be false.

Reply:
@kjohnson I updated variable name to `$is_validating_donation_form_on_ajax`.
ref: https://github.com/impress-org/givewp/compare/cb5aa349e381...ffa7622bc1e9 https://github.com/impress-org/givewp/compare/ffa7622bc1e9...5b152ce990f0